From f62aa5fee00cea148a6bbd8e13c284c8f7cbe121 Mon Sep 17 00:00:00 2001
From: "G.h. Gao" <guanhua.gao@nxp.com>
Date: Fri, 14 Dec 2018 10:41:06 +0800
Subject: [PATCH 762/767] staging: fsl-dpaa2/mac: fix the if_mode type

The of_get_phy_mode function returns a signed integer and should be
treated as such. Since the underlying type for the enum cannot be
determined (that's up to the compiler), it can be signed or unsigned,
there may be cases when the comparison to 0 is always true.

Fix this by making the if_mode type integer.

should be signed since of_get_phy_mode could return negative
also

Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 38dc66a2b33d..8b4c330a2877 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -554,7 +554,7 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 	struct dpaa2_mac_priv	*priv = NULL;
 	struct device_node	*phy_node, *dpmac_node;
 	struct net_device	*netdev;
-	phy_interface_t		if_mode;
+	int			if_mode;
 	int			err = 0;
 
 	dev = &mc_dev->dev;
-- 
2.17.0

